### PR TITLE
FIX don't remove jobs when processing via publish job

### PR DESCRIPTION
**Pull request by wilr**
WorkflowPublishTargetJob::process() triggers a onBeforeWrite() which in turn can lead to job descriptors being deleted while still being used.

### CODE REVIEW

**Comment by michalkleiner** [comment]
This works well in my opinion, what do you think about the change @GuySartorelli? Can retarget 5 and make it a new minor, or keep as a fix since the jobs really shouldn't be deleted by the publish/unpublish action itself.

File: src/Extensions/WorkflowEmbargoExpiryExtension.php [hunk 1]
```diff
@@ -195,7 +195,7 @@ public function clearUnPublishJob()
     {
         // Cancel any in-progress unpublish job
```
Comment (ChloeSartorelli) [inline src/Extensions/WorkflowEmbargoExpiryExtension.php:196]: This comment seems to indicate that we're meant to explicitly be deleting jobs that aren't finished? So presumably the other method is also doing that intentionally?
I haven't looked any more deeply than that, I was just having a quick look to answer the question of which branch to target

**Review by GuySartorelli — COMMENTED** [review]
With the change as is I'd be happy with it being tagged as a patch release, so this branch is fine.
I do have a question below but as stated there I haven't looked deeply into it - I'll trust both of your judgment there.

File: src/Extensions/WorkflowEmbargoExpiryExtension.php (hunk 1 above)
Comment (michalkleiner) [reply src/Extensions/WorkflowEmbargoExpiryExtension.php:196]: Thanks for taking the time @GuySartorelli.

Can you maybe rephrase your comment? I'm not sure I fully get which other method you mean and what you think should or shouldn't be happening.

Besides that, I think there's definitely room for improvement around the jobs handling within the module here, this patch is a step forward but a proper review of what jobs are created, executed, removed, updated etc. would still be beneficial. I reckon there were enhancements in the queued jobs module over time that could be leveraged here (just a gut feeling, no specifics in mind).

Comment (ChloeSartorelli) [reply src/Extensions/WorkflowEmbargoExpiryExtension.php:196]: Sorry yeah.

This comment is in `clearUnPublishJob()` - `clearPublishJob()` is very similar and is the other method that was altered in this pr, though it lacks a comment saying anything about jobs that are still running. I'm guessing they're intended to function the same way as one another, so if this method is meant to delete unfinished jobs, the publish method probably is as well.

I'm not sure what should or shouldn't be happening - I just noticed that this comment in the code seems to be contrary to the change being made. I haven't looked at what this functionality is doing, or when it is calle,d or what the side effects ,ight be etc as that wasn't part of the type of review I was doing. I'll leave it to you and @wilr to make determinations as to what should be happening here.

Comment (michalkleiner) [reply src/Extensions/WorkflowEmbargoExpiryExtension.php:196]: Yep, those methods should behave in a similar way. The publish/unpublish are just two variants of the same action and the jobs should behave the same. I'd say the other method just doesn't have the comment (rather than being different in principle).

**Review by michalkleiner — APPROVED** [review]
Ok for  me and it runs on an MBIE site for nearly 2 years now so I guess it's safe to merge.
Any objections @GuySartorelli?

**Comment by GuySartorelli** [comment]
No objections here.
